packages/react-reconciler/src/ReactFiberWorkLoop.js

```diff
@@ -2176,6 +2178,19 @@ export function flushPassiveEffects() {
   }
 }
 
+export function enqueuePendingPassiveEffectDestroyFn(
+  destroy: () => void,
+): void {
+  if (deferPassiveEffectCleanupDuringUnmount) {
+    pendingUnmountedPassiveEffectDestroyFunctions.push(destroy);
+    rootDoesHavePassiveEffects = true;
+    scheduleCallback(NormalPriority, () => {
```

Comment:
This will schedule a callback for every clean-up effect. Need to check if `rootDoesHavePassiveEffects` is already true.

Reply:
Ah, good call!